fake_switches/juniper/juniper_netconf_datastore.py

```diff
@@ -210,10 +217,11 @@ def parse_interfaces(self, conf, etree_conf):
             operation = resolve_operation(interface_node)
             if operation == "delete" and isinstance(port, AggregatedPort):
                 conf.remove_port(port)
+            if operation == "delete" and not isinstance(port, AggregatedPort) and not isinstance(port, VlanPort):
```

Comment:
Nope, can't make a test that references a Vlan Port, since we can't create one, the port matching would just return nothing... the best would probably be to have a PhysicalPort and have no base Port (would be an abstract class)... but that might be a refactor for some other time